extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts

```diff
@@ -172,4 +189,33 @@ export class DeployAzureSQLDBWizard extends WizardBase<DeployAzureSQLDBWizard, W
 			level: azdata.window.MessageLevel.Error
 		};
 	}
+
+	public async refreshPages() {
+
+		const pageCount = this.wizardObject.pages.length;
+		for (let i = 1; i < pageCount; i++) {
+			this.wizardObject.removePage(this.wizardObject.pages.length - 1);
+			this.wizardObject.pages.pop();
+		}
+		if (instanceOfAzureSQLDBDeploymentProvider(this.resourceProvider)) {
+			this._wizardInfo = this.resourceProvider.azureSQLDBWizard!;
+		} else {
+			return;
+		}
+
+		const newPages = this.getPages();
+
+		newPages[0] = this.pages[0];
```

Comment:
So you create the new EULA page, only to immediately overwrite it? That doesn't seem great

Reply:
Probably I can send an additional flag to the method. To avoid it

Reply:
Why is this necessary in the first place? What are you trying to accomplish by not recreating it like all the others?

Reply:
This is to save the state of the page.

Reply:
Ok - but why is this page special? Why aren't we saving the state of all the pages? When does this get called anyways?

Reply:
This is called when we change the deployment options. After changing the options, the provider will change requiring us to fetch pages for that provider. When this happens, all the pages in the wizard gets removed except the first one (Tools and Eula Page) which helped us make that change. 

For reference, it is this component under Tools And Eula Page. 
![image](https://user-images.githubusercontent.com/6816294/96614261-c7c63100-12b4-11eb-8087-980ce86b70f6.png)

Reply:
So really it seems like the tools and EULA page needs to be a separate thing at the resourceDeploymentTypes level instead of the providers level because we haven't even chosen a provider yet.

We have a code structuring issue right now in that when we launch the wizard we currently assume that we know the provider (because in the old format we did). But that's not the case anymore. There should be a generic "wizard" class that starts off with the Tools and EULA page only and then when a provider is selected it will create the other pages separately. 

Stuff like this is why I wanted to discuss more before going and trying to fit this into the existing logic - we're fundamentally changing how these wizards are launched and thus trying to keep the logic to match with the old way is probably going to be more effort and result in messier code than just "starting over" with a new base wizard type that can handle different providers.